frontend/src/components/shared/WithItemRenderComponent.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { Container, FetchedObject, Index, Individual, ItemsByID, Project, Sample, User } from '../../models/frontend_models'
+import { selectContainersByID, selectIndicesByID, selectIndividualsByID, selectProjectsByID, selectSamplesByID, selectUsersByID } from '../../selectors'
+import { createWithItem, withContainer, withIndex, withIndividual, withProject, withSample, withUser } from '../../utils/withItem'
+
+/**
+ * WithItemRenderComponent
+ * 
+ * WithItemRenderComponent is used where we need to load an object from the backend if it
+ * is not already in the redux store. It handles loading the object and when it is ready
+ * it passes the object to a render function that you provide. 
+ * 
+ * There specific WithXXXRenderComponent components for samples, containers, etc..
+ * 
+ * The component takes these props:
+ * 
+ * objectID: The ID of the object you want to render
+ * render: A function that takes the object as a parameter and returns a React.Element.=
```

Comment:
Something i wonder if you are trying to trigger my OCD ... = after . here. Missing . last sentence. etc ... a bit earlier ...

Reply:
Sorry, I let ChatGPT write that comment for me.